docs/book/src/reference/versions.md

```diff
@@ -78,6 +78,24 @@ The Kubeadm Control Plane Provider talks to the API server and etcd members of e
 
 The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 
+#### clusterctl
+
+It is strongly recommended using a version of [clusterctl](../clusterctl/overview.md) that exactly matches the version
+of the Cluster API core provider; in case of upgrades, clusterctl should be upgraded first and used to upgrade
+all the other components.
```

Comment:
Not sure what we're trying to communicate here, do we want users to upgrade clusterctl to the latest available version all the time?

Reply:
Yes, IMO this is the safest approach: upgrade clusterctl first, then use the latest clusterctl to upgrade other components

Reply:
```suggestion
It is strongly recommended using the latest version of [clusterctl](../clusterctl/overview.md) before attempting to upgrade your clusters.
```

How about something like this? I'd also open an issue if we don't have one already to have ways in clusterctl to check for its latest version

Reply:
Thinking back to this might be I have misunderstood the original question because the consideration does not apply to upgrades only .

I have in mind some recent use cases of uses trying to init a cluster with version 0.3.7 but using an older version of clusterctl, and failing due to the changes for envsubst; I do expect might happen in the near future as well, e.g. for the changes on cert-manager, or for the changes on the move logic.

This is why IMO the version of clusterctl should always be >= to the version of CAPI installed in the cluster (and as corollary, clusterctl should be updated before all the other components)

Reply:
Yes. Ideally, the user should always update to the latest version of clusterctl before doing any operations in order to avoid unintended side effects. 
We had a discussion regarding clusterctl doing a version check ([see slack thread](https://kubernetes.slack.com/archives/C8TSNPY4T/p1594765826104200?thread_ts=1594765139.103900&cid=C8TSNPY4T)). 

I can create an issue regarding clusterctl doing a version check.

Reply:
I created this issue regarding above: https://github.com/kubernetes-sigs/cluster-api/issues/3466

Reply:
I have aligned the doc to the issue above, please PTAL